Group 8 Final Project Proposal

ajwhitney23 opened the pull request.

ajwhitney23 committed:
added proposal

charlieroberts commented:
OK! Can you give me a better sense of what you'd like to do differently from the existing project? It'd be great to hear alternative UI ideas etc if you have them. Or are you planning on mimicking the existing interface?

Otherwise, this seems very doable and like it will be a great opportunity to dig in and learn more about React + Typescript.

ajwhitney23 committed:
added starter server code

NickFrangie commented:
> OK! Can you give me a better sense of what you'd like to do differently from the existing project? It'd be great to hear alternative UI ideas etc if you have them. Or are you planning on mimicking the existing interface?
> 
> Otherwise, this seems very doable and like it will be a great opportunity to dig in and learn more about React + Typescript.

We want to make a number of UX changes that alter how the user interacts with the program. Some examples include: 
- Change party members once they are selected (currently, you have to delete and add new party members to enact changes).
- Allow users to select Artifacts and Weapons disassociated from characters (this will allow users greater control of what they want to use the program for).
- Cleaner UI display (less sliders, separate togglables into 2 distinct displays for hiding items vs. selecting options).
- Reset Data should be displayed prominently with a warning prior to wiping user data

NickFrangie committed:
Folder Configuration

NickFrangie committed:
Setup React with Server

ajwhitney23 committed:
fixed db connection issues

ajwhitney23 committed:
fix up replace

ajwhitney23 committed:
Merge pull request #1 from ajwhitney23/react

Setup React with Server

NickFrangie committed:
Setup skeleton code

ajwhitney23 committed:
Merge branch 'main' into backend

ajwhitney23 committed:
removed debug code

ajwhitney23 committed:
Merge pull request #2 from ajwhitney23/backend

Setup skeleton code

ajwhitney23 committed:
fetch pref data for user w/ ID 1 for now, until login is implemented

ajwhitney23 committed:
remove console.log

ajwhitney23 committed:
updated server to parse json when received

lockeanddemosthenes committed:
Added images from Genshin

All spaces are replaced with "_" and all apostrophes are removed (e.g. "Gladiators_Finale.png"). Folders and their contents are as follows:

artifacts - only flowers from each set; name format is "[setname].png" (e.g. "Brave_Heart.png")
chara - a thumbnail for each character; name format is "Character_[charname]_Thumb.png" (e.g. ("Character_Kaedehara_Kazuha_Thumb.png")
charaasc - character ascension materials; name format is "Item_[itemname].png" (e.g. "Item_Calla_Lily.png")
commonmats - materials used in at least two of character ascension, talent upgrade, and weapon ascension; name format is "Item_[itemname].png" (e.g. "Item_Chaos_Oculus.png")
elements - element icons; name format is "Element_[elementname].png" (e.g. "Element_Anemo.png")
misc - mora, level-up, and weapon enhancement; name format is "Item_[itemname].png" (e.g. "Item_Heros_Wit.png")
talent - talent upgrade materials; name format is "Item_[itemname].png" (e.g. "Item_Bloodjade_Branch.png")
weapon - weapons; name format is "Weapon_[weaponname].png" (e.g. "Weapon_AmosBow.png")
weaponasc - weapon ascension materials; name format is "Item_[itemname].png" (e.g. "Item_Boreal_Wolfs_Nostalgia.png")
weapontype - weapon type icons; name format is "Icon_[weptype].png" (e.g. "Icon_Claymore.png")

lockeanddemosthenes committed:
Merge branch 'main' of https://github.com/ajwhitney23/final_project into main

lockeanddemosthenes committed:
Merge pull request #3 from ajwhitney23/dbinteraction

fetch pref data for user w/ ID 1 for now, until login is implemented

lockeanddemosthenes committed:
Merge branch 'main' of https://github.com/ajwhitney23/final_project into main

ajwhitney23 committed:
Merge pull request #4 from ajwhitney23/images

Images

ajwhitney23 committed:
updated server to use session id, added login button, had to set redirect to http://localhost since sendfile was setting to https

ajwhitney23 committed:
updated server and mainroute for user handling

ajwhitney23 committed:
removed reference to localhost

ajwhitney23 committed:
removed localhost references in react

ajwhitney23 committed:
changed logout to not redirect

ajwhitney23 committed:
remove login.md

ajwhitney23 committed:
Merge pull request #5 from ajwhitney23/login

updated server to use session id, added login button, had to set redi…

ajwhitney23 committed:
updated login to init char/weapon/artifact prefs along with added button to save prefs into db

lockeanddemosthenes committed:
Merge pull request #6 from ajwhitney23/loadonlogin

updated login to init char/weapon/artifact prefs along with added but…

lockeanddemosthenes committed:
moved all the images into static/img

lockeanddemosthenes committed:
image directory fixed

i am a fool, montag.

NickFrangie committed:
addPref Functionality

lockeanddemosthenes committed:
wowah menu button

lockeanddemosthenes committed:
slight pretty changes

NickFrangie committed:
removePref, updatePref Functionality

NickFrangie committed:
Demo functionality

ajwhitney23 committed:
Merge pull request #8 from ajwhitney23/select-helpers

Select Helpers

ajwhitney23 committed:
wip

lockeanddemosthenes committed:
moved menucontainer into mainroute

ajwhitney23 committed:
farming display now sets days farm and locations

NickFrangie committed:
Reorganize HTML order & file structure

NickFrangie committed:
Merge branch 'main' into flyout

NickFrangie committed:
Merge pull request #7 from ajwhitney23/flyout

Fix img location and added menu

ajwhitney23 committed:
adding farming display, needs api data to test

ajwhitney23 committed:
Merge branch 'main' into calendar

Commit:
Added json files from database, started interface

ajwhitney23 committed:
working calendar, needs data

lockeanddemosthenes committed:
Merge pull request #9 from ajwhitney23/calendar

Calendar

ajwhitney23 committed:
automatically updates db

ajwhitney23 committed:
now automatically updates, and cleanup/ error prevention

ajwhitney23 committed:
added a reset button

lockeanddemosthenes committed:
all img names fixed, added some missing imgs, removed unnecessary imgs

lockeanddemosthenes committed:
farmables folder

lockeanddemosthenes committed:
Merge pull request #10 from ajwhitney23/automaticallysave

automatically updates db

NickFrangie committed:
Merge pull request #11 from ajwhitney23/imgNameFix

all img names fixed, added some missing imgs, removed unnecessary imgs

Commit:
ArtifactProps complete, issues with accessing

NickFrangie committed:
SelectPrefButton added

NickFrangie committed:
SelectPrefButton content def to children

AstroSeer committed:
Merge pull request #12 from ajwhitney23/select-buttons

Select buttons

AstroSeer committed:
Merge pull request #13 from ajwhitney23/main

Wouldnt pull main files into new branch

ajwhitney23 committed:
updated to most recent main

ajwhitney23 committed:
updated data to condense weapons and compile

ajwhitney23 committed:
updated data to be fetched

lockeanddemosthenes committed:
calendar display pretty :)

lockeanddemosthenes committed:
commented out dummy data

ajwhitney23 committed:
loading json

ajwhitney23 committed:
debug prints

AstroSeer committed:
Menu has working button to close

ajwhitney23 committed:
Merge pull request #14 from ajwhitney23/farmingdisplay

calendar display pretty :)

AstroSeer committed:
Update Menu.jsx

AstroSeer committed:
Merge branch 'menu-button'

ajwhitney23 committed:
Merge branch 'main' into api

NickFrangie committed:
Revert "Merge branch 'main' into api"

This reverts commit be8439c414935418710875a3e4672053abe4f181, reversing
changes made to 0449419c3501cfaa12c14d5500974af5ff0b699b.

ajwhitney23 committed:
fixed imports

ajwhitney23 committed:
updates to debug

ajwhitney23 committed:
removed farmable props

ajwhitney23 committed:
forgot one spot

ajwhitney23 committed:
updated locations

NickFrangie committed:
Used Setters

ajwhitney23 committed:
updated api/select to now functional state

NickFrangie committed:
Merge pull request #16 from ajwhitney23/api

Merge API Changes into Main

ajwhitney23 committed:
update filters to now properly show data

NickFrangie committed:
Merge pull request #17 from ajwhitney23/dayfilters

update filters to now properly show data

NickFrangie committed:
Fixed Button Display on Reload

NickFrangie committed:
Merge branch 'main' into select-button-bf

ajwhitney23 committed:
Merge pull request #18 from ajwhitney23/select-button-bf

Fixed Button Display on Reload

NickFrangie committed:
Readded window redirect

ajwhitney23 committed:
andrew got smarted today

NickFrangie committed:
SelectButtons work after refresh

NickFrangie committed:
Logout Redirect

NickFrangie committed:
Merge pull request #19 from ajwhitney23/out-of-sync

Fixed out of sync issues

lockeanddemosthenes committed:
all img names fixed

lockeanddemosthenes committed:
display names fixed

lockeanddemosthenes committed:
add weapon & element to characters

lockeanddemosthenes committed:
favicon???? may need to be fixed

lockeanddemosthenes committed:
adjusted brightness of weapontype icon

lockeanddemosthenes committed:
fixed color of weapontype icon

lockeanddemosthenes committed:
make images in CharacterSelect a little bigger

lockeanddemosthenes committed:
Revert "make images in CharacterSelect a little bigger"

This reverts commit 26379ace28eba82e0a9d4437dc594ee211690106.

lockeanddemosthenes committed:
Revert "Revert "make images in CharacterSelect a little bigger""

This reverts commit 9f8e271d5bea428d6b1d228a267c815c5ca84d5d.

lockeanddemosthenes committed:
we are out here, making things pretty.
ISSUE: names that go on two lines (most notable in weapons) will make the button drift down about 15 pixels.

AstroSeer committed:
Tabs in Menu

AstroSeer committed:
Added components to tabs

ajwhitney23 committed:
Merge branch 'main' into menu-button

ajwhitney23 committed:
updated menu to pass in tsx

ajwhitney23 committed:
minimal viable product

ajwhitney23 committed:
update formatting

ajwhitney23 committed:
update mainroute to remove comment code

Commit:
package.json change

Commit:
Merge branch 'menu-button' of https://github.com/ajwhitney23/final_project into menu-button

ajwhitney23 committed:
Merge pull request #20 from ajwhitney23/menu-button

Menu button

ajwhitney23 committed:
Merge branch 'main' into display-updates

NickFrangie committed:
Format Readme for Assignment

ajwhitney23 committed:
andrew whitney readme update

NickFrangie committed:
Nick sections filled

lockeanddemosthenes committed:
some styling for selection now that the menu exists

lockeanddemosthenes committed:
Errica sections filled

ajwhitney23 committed:
updated andrew whitney readme

ajwhitney23 committed:
Merge branch 'readme' of github.com:ajwhitney23/final_project into readme

ajwhitney23 committed:
updated readme

lockeanddemosthenes committed:
started answering question 1

lockeanddemosthenes committed:
Merge branch 'readme' of https://github.com/ajwhitney23/final_project into readme

lockeanddemosthenes committed:
HOT SCROLLBAR FRESH OFF THE KBBQ GRILL

lockeanddemosthenes committed:
made tabs pretty lol

lockeanddemosthenes committed:
slight img padding so i stop screaming and hitting my head

AstroSeer committed:
Matt's section added and first part additions

ajwhitney23 committed:
moving button over one

ajwhitney23 committed:
Merge pull request #21 from ajwhitney23/display-updates

Display updates

Commit:
Menu pushing works

lockeanddemosthenes committed:
formatting updates i think

lockeanddemosthenes committed:
more formatting...

Commit:
Merge branch 'main' of https://github.com/ajwhitney23/final_project into menu-push

AstroSeer committed:
Merge pull request #22 from ajwhitney23/menu-push

Menu push

lockeanddemosthenes committed:
SPELLING ERROR RAAAH

Commit:
Michael ~ Added my section

lockeanddemosthenes committed:
teeny tiny menu button changes

ajwhitney23 committed:
Merge pull request #23 from ajwhitney23/menu-butt

teeny tiny menu button changes

ajwhitney23 committed:
added link

ajwhitney23 committed:
update 2

NickFrangie committed:
Video Link Added

ajwhitney23 committed:
Merge pull request #24 from ajwhitney23/readme

Readme
